server/mqtt.go

```diff
@@ -1701,9 +1705,17 @@ func (jsa *mqttJSA) createDurableConsumer(cfg *CreateConsumerRequest) (*JSApiCon
 	return ccr, ccr.ToError()
 }
 
-func (jsa *mqttJSA) deleteConsumer(streamName, consName string) (*JSApiConsumerDeleteResponse, error) {
+// if noWait is specified, does not wait for the JS response, returns nil
+func (jsa *mqttJSA) deleteConsumer(streamName, consName string, noWait bool) (*JSApiConsumerDeleteResponse, error) {
 	subj := fmt.Sprintf(JSApiConsumerDeleteT, streamName, consName)
-	cdri, err := jsa.newRequest(mqttJSAConsumerDel, subj, 0, nil)
+
+	timeout := mqttJSAPITimeout
```

Comment:
I checked that in all other places we use an explicit value. 1/5 this is intention-revealing enough, no need for another flag.